### PR TITLE
support fastify 5

### PR DESCRIPTION
**Pull request by danielwitz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,45 +18,45 @@
...
-    "jest": "^28.1.1",
-    "lint-staged": "^13.0.2",
-    "prettier": "^2.7.1",
-    "ts-jest": "^28.0.5",
-    "ts-node": "^10.8.1",
-    "typescript": "^4.7.4"
+    "@osskit/tsconfig": "^0.0.7",
+    "@types/http-errors": "^2.0.4",
+    "@types/node": "^22.7.5",
+    "@typescript-eslint/eslint-plugin": "^8.9.0",
```
Comment (noamokman) [inline package.json:43]: No need

Comment (danielwitz) [reply package.json]: done

File: package.json [hunk 2]
```diff
@@ -18,45 +18,44 @@
...
-    "jest": "^28.1.1",
-    "lint-staged": "^13.0.2",
-    "prettier": "^2.7.1",
-    "ts-jest": "^28.0.5",
-    "ts-node": "^10.8.1",
-    "typescript": "^4.7.4"
+    "@osskit/tsconfig": "^0.0.7",
+    "@types/http-errors": "^2.0.4",
+    "@types/node": "^22.7.5",
+    "@typescript-eslint/parser": "^8.9.0",
```
Comment (noamokman) [inline package.json:43]: Also

File: package.json [hunk 3]
```diff
@@ -18,45 +18,44 @@
...
+    "@vitest/ui": "^2.1.3",
+    "eslint": "^9.12.0",
+    "eslint-plugin-import": "^2.31.0",
```
Comment (noamokman) [inline package.json:46]: .

File: package.json [hunk 4]
```diff
@@ -18,45 +18,44 @@
...
+    "eslint-plugin-unicorn": "^56.0.0",
```
Comment (noamokman) [inline package.json:47]: .

File: package.json [hunk 5]
```diff
@@ -18,45 +18,44 @@
...
+    "fastify": "^5.0.0",
+    "husky": "^9.1.6",
+    "lint-staged": "^15.2.10",
+    "prettier": "^3.3.3",
+    "ts-node": "^10.9.2",
```
Comment (noamokman) [inline package.json:52]: Not needed

File: tsconfig.eslint.json
Comment (noamokman) [inline tsconfig.eslint.json:1]: Actually not needed, also simply the eslint config not to use it

File: eslint.config.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import typescriptEslint from 'typescript-eslint';
+
+export default typescriptEslint.config(
+  {
+    ignores: ['dist/**', '**/dist/**'],
+  },
+  {
+    ...baseConfig,
+    files: ['src/**/*.ts', 'tests/**/*.ts'],
+    languageOptions: {
```
Comment (noamokman) [inline eslint.config.js:11]: No need for this now

**Review by noamokman — APPROVED** [review]

**Pull request merged**